### ISSUE TITLE
Icon/logo

### ISSUE THREAD

**Issue by amezin** [ddterm/gnome-shell-extension-ddterm#724]
It would be nice if ddterm had it's own icon, not the generic terminal icon from system theme.

Unfortunately, I do not have the skills to draw a good looking icon.

**Comment by LukIsHere** [comment]
how about :
![ddterm](https://github.com/ddterm/gnome-shell-extension-ddterm/assets/92756992/49b36864-6403-49ec-9dc8-decd0226bf48)

**Comment by LukIsHere** [comment]
![concept](https://github.com/ddterm/gnome-shell-extension-ddterm/assets/92756992/895d48ed-6278-4c06-a15f-daf79f398f54)

**Comment by amezin** [comment]
Ironically, it shows a feature ddterm doesn't have https://github.com/ddterm/gnome-shell-extension-ddterm/issues/129

But looks good otherwise, thanks.

There are also places where a colored, non-symbolic icon is expected -  for example, application grid: https://github.com/ddterm/gnome-shell-extension-ddterm/blob/master/docs/screenshots/desktop-entry.png Do you have anything for such cases?

**Comment by LukIsHere** [comment]
a concept
![ddtermConcept](https://github.com/ddterm/gnome-shell-extension-ddterm/assets/92756992/8aaada18-b41d-42cc-bc29-34550475feba)
can probably remake it in similar style to this
![Zrzut ekranu z 2024-03-19 12-49-35](https://github.com/ddterm/gnome-shell-extension-ddterm/assets/92756992/cccbd92c-9e3e-4ae1-b605-cf5c1df29164)

**Comment by amezin** [comment]
Also the icon looks misaligned - not centered vertically, moved too much to the bottom. Even if it's intentional - I'm not sure it's a good idea.

When you have both variants of the icon ready, could you make a pull request, adding the files to the repository? Don't worry about fixing the code to load/use the icons - it might be non-trivial, I can do it later. But you may try if you want.

**Comment by thenameisluk** [comment]
### Hi
it's been a while (sorry), i completely forgot about this
but 've manage to make actual gnome style logo

### three variants
no background
<img src="https://github.com/user-attachments/assets/71de6b68-ef02-4eff-ba0f-c2eddcfbf662" width="124">

black background
<img src="https://github.com/user-attachments/assets/2776034f-defc-4bee-8ec1-bb2fd8bfe471" width="124">

white background
<img src="https://github.com/user-attachments/assets/98f434d7-95d1-49ff-a81d-0dac825dc4ea" width="124">


### also source file
[ddterm.xcf.gz](https://github.com/user-attachments/files/18218809/ddterm.xcf.gz)
can be opened in gimp to scale up, change colors, move stuff around, change background, etc.
_Note. compressed it with gzip since github doesn't accept uploading .xcf files, can be unpackeg with ```gunzip```_

fell free to use it, i don't really want to make pr

about the topbar icon, i think the currently used one is fine, no need to add unnecessary details

**Comment by amezin** [comment]
Ok could you let me know:

- The license you prefer. GNOME's symbolic icon set (which I already use) is available under CC0 https://creativecommons.org/publicdomain/zero/1.0/ - are you ok with applying the same license to this icon? CC BY-SA 4.0 https://creativecommons.org/licenses/by-sa/4.0/ is another option

- How should I mention you in the credits and file copyright notice? (there will be an "About" dialog... one day)

**Comment by thenameisluk** [comment]
- CC0 is fine with me
- if you are asking about what to put, you can put thenameisluk (if you wan for whatever reason to include email lukasz.osadik@wp.pl/website https://iaml.uk)

**Comment by amezin** [comment]
There's another issue: svg is currently the de-facto standard for icons. I tried automatic converstion to svg, but the result was honestly trash. Bundling lots of re-scaled pngs of various sizes takes a lot of space - literally half of the size of the package.

So for now I'm keeping the terminal icon from the system icon set.

**Comment by thenameisluk** [comment]
![Image](https://github.com/user-attachments/assets/f90885a6-c025-410e-874d-308c5d4938d3)
does this work?

[ddterm.svg.gz](https://github.com/user-attachments/files/20022800/ddterm.svg.gz)

**Comment by thenameisluk** [comment]
@amezin ?

**Comment by amezin** [comment]
Yes, it works. However, the icon looks a bit out of place - there's a bit of mismatch with the surrounding icons.

Ubuntu:

![Image](https://github.com/user-attachments/assets/4e08588d-d42b-4679-a8c8-dac5be0c9c9a)

Stock GNOME (Alpine, but the distro shouldn't really matter here):

![Image](https://github.com/user-attachments/assets/28b04fac-c50d-40c6-b7b5-6d9a584e0d2d)

Probably this can be solved by tweaking the colors (it looks like other icons have more contrast/saturation), and making it a bit more square.

**Comment by amezin** [comment]
Here's the package with the icon, you can try it yourself: [ddterm@amezin.github.com.shell-extension.zip](https://github.com/user-attachments/files/20060178/ddterm%40amezin.github.com.shell-extension.zip)

If you want to modify the icon, edit the one in `~/.local/share/gnome-shell/extensions/ddterm@amezin.github.com/data/`, then disable and re-enable the extension

**Comment by thenameisluk** [comment]
ubuntu does their own thing with their icon pack, no other distro i know uses it so i woundn't worry about it

but yeah it looks a bit out of place when it comes to stock gnome
imma try messing around with colors to make it look better

**Comment by amezin** [comment]
> ubuntu does their own thing with their icon pack, no other distro i know uses it so i woundn't worry about it

Ubuntu is a popular distro, so it would be nice to have an icon which looks well in that environment too. Ubuntu's icon design is usually not that far from stock GNOME's

> imma try messing around with colors to make it look better

https://flathub.org/apps/org.gnome.design.Palette and https://flathub.org/apps/org.gnome.design.AppIconPreview

**Comment by thenameisluk** [comment]
in my honest opinion ubuntu icons are flat compared to gnome which is isometric
i'll make sure it is alligned with gnome and by product might be it looking fine on ubuntu but designing stuff around one distro doesn't seam like a great idea (i doubt there are that many users of this extension on ubuntu, it's widely considered "new user friendly" and using a terminal isn't really something new users often want to do)

either ways thanks for the links
definitely gonna use them

**Comment by thenameisluk** [comment]
![Image](https://github.com/user-attachments/assets/49a83b1d-d02d-493b-afe3-5b55aa469c0d)
[com.ddterm.reg.svg.gz](https://github.com/user-attachments/files/20070832/com.ddterm.reg.svg.gz)

![Image](https://github.com/user-attachments/assets/84ee5aee-14e7-4da9-94ec-96dca64007e7)

this one follows "the gnome grid" and uses gnome color pallet
thoughts?

also source:
[com.ddterm.svg.gz](https://github.com/user-attachments/files/20070831/com.ddterm.svg.gz)
(it might look like the one above but can be opened in gnome icon preview app to export symbol/regular/nightly)

**Comment by amezin** [comment]
Yes, colors seem to be correct now. But now that thing on the top doesn't look like GNOME Shell's top bar at all

![Image](https://github.com/user-attachments/assets/a3a00551-b5f5-426a-a3c2-c5430fc5308b)

**Comment by thenameisluk** [comment]
how about
![Image](https://github.com/user-attachments/assets/fcbb3281-b87f-4166-a9de-17e82576c355)
[com.ddterm.reg.svg.gz](https://github.com/user-attachments/files/20076727/com.ddterm.reg.svg.gz)
this?
![Image](https://github.com/user-attachments/assets/3d6930d8-643d-4521-99cb-690f001f4f05)

i would avoid adding too much details (like clock or volume/power icons) since it would go against [GNOME humane interface guidelines](https://developer.gnome.org/hig/guidelines/app-icons.html#Detail)
the classic gnome bean should be iconic enough to convey metaphore

**Comment by amezin** [comment]
What do you think about:

- Reducing the height of the "top bar" a bit? Maybe without making the "Activities button" smaller, just reducing "empty" space.
- Making the "Activities button" flat - like it is on the real panel?
- Making the dot in the "Activities button" smaller and darker (like it is in reality when you have no additional workspaces)? Also see https://github.com/GNOME/gnome-shell/blob/main/data/icons/scalable/actions/shell-focus-top-bar-symbolic.svg
- Removing rounded corners from the "top bar"? Real one doesn't have them. Not every corner everywhere has to be round
- Replacing ">_" with ">dd" or something like that?

Also, the fact that the window is narrower than the top bar still bothers me. Because it advertises a feature that does not exist/does not work - https://github.com/ddterm/gnome-shell-extension-ddterm/issues/129

**Comment by amezin** [comment]
> all gnome icons have rounded corners

https://gitlab.gnome.org/World/design/icon-library/-/blob/master/data/icons/org.gnome.design.IconLibrary.svg

Also Contacts, Devhelp, Dictionary

> wouldn't that be too much detail?

`>_` is 2 characters, `>dd` is 3 characters. There's no need to reduce their size, there is enough space already - so downscaling shouldn't be a problem.

> sure (but not fully convinced is needed/looks good, the icon is meant to convey an idea, not represent the looks of the program)

Yeah, it now looks like window title bar more than Shell's top bar, probably not a great idea.

**Comment by thenameisluk** [comment]
> https://gitlab.gnome.org/World/design/icon-library/-/blob/master/data/icons/org.gnome.design.IconLibrary.svg
> 
> Also Contacts, Devhelp, Dictionary

do not rounded corners look better?

![Image](https://github.com/user-attachments/assets/5e2ceb7a-329d-40f1-bb0a-d5e60be1b0fb)
(sharp)

![Image](https://github.com/user-attachments/assets/6410e595-f69e-49fb-abfb-0652e13a5d0d)
(rounded)

> `>_` is 2 characters, `>dd` is 3 characters. There's no need to reduce their size, there is enough space already - so downscaling shouldn't be a problem.

you need to reduce their thickness in order to not make them a blob

![Image](https://github.com/user-attachments/assets/0d80144e-11de-4c34-87ef-a4db35bd1afc)
(blob)

![Image](https://github.com/user-attachments/assets/5e2ceb7a-329d-40f1-bb0a-d5e60be1b0fb)
(reduced thickness)

![Image](https://github.com/user-attachments/assets/68916864-2aa5-4670-9989-eb62a53c91da)
(classic)

does it look good?

**Comment by thenameisluk** [comment]
imma stop attaching source for each variand cause it's getting hard to manage

**Comment by thenameisluk** [comment]
i don't think these scale well compared to classic representation

![Image](https://github.com/user-attachments/assets/432819dd-03bd-46c8-9931-d92310ad6089)

![Image](https://github.com/user-attachments/assets/73178420-67e5-4906-93ca-c1b158464e89)

![Image](https://github.com/user-attachments/assets/14d64221-b2a3-4d3f-928e-a65d77bd6de2)

also it's not far from adwaita mono
![Image](https://github.com/user-attachments/assets/230773ab-1274-4d31-8ddd-29b511afd50b)

**Comment by amezin** [comment]
I like this 

![Image](https://github.com/user-attachments/assets/db3dbf7f-e95f-4d51-a485-dbcf82dfd3b8)

more than this

![Image](https://github.com/user-attachments/assets/a08a3bcd-d306-42ad-86f9-8d8e722bf940)

And `dd` letters help me understand which of the installed terminal apps it is immediately, without any additional labels.

Maybe something "in the middle" would be even better? I.e. make `>` a bit larger than on the first picture, but not as large as on the last one. So that on the small icon you'll still clearly see `>` at least

**Comment by thenameisluk** [comment]
> And `dd` letters help me understand which of the installed terminal apps it is immediately, without any additional labels.

not a single gnome terminal known to me has top of the shell attached to it (do you have any examples?)

![Image](https://github.com/user-attachments/assets/1d910dc9-1171-43d9-aac0-ed7e0761e330)

![Image](https://github.com/user-attachments/assets/a3e3652f-56ff-4ee7-b71a-809e22a37cde)

this is a terminal that drops down from the top

![Image](https://github.com/user-attachments/assets/34c65594-a8fb-443a-9e5b-106ad17d2bf4)

there is no need to force letters onto the icon that is easily distinguishable

![Image](https://github.com/user-attachments/assets/000c1add-854b-4794-a899-9421d44e9524)

"**Avoid** metaphors that are unrelated to the name or function of the app, such as using **characters** or mascots. Also avoid using logos which rely on a specific visual style."

 
> Maybe something "in the middle" would be even better? I.e. make `>` a bit larger than on the first picture, but not as large as on the last one. So that on the small icon you'll still clearly see `>` at least

even if i make it larger it doesn't scale well

![Image](https://github.com/user-attachments/assets/b8b84af9-1d36-4ff5-bc55-5783b6199a5f)

and ofc we can make text even bigger but at some point it just stops looking good

![Image](https://github.com/user-attachments/assets/685022af-5a1f-4567-9d0e-7826ea1ea9c6)

![Image](https://github.com/user-attachments/assets/72a619bd-96cb-404c-9cc7-d7a11a84f8b3)

**Comment by amezin** [comment]
> not a single gnome terminal known to me has top of the shell attached to it (do you have any examples?)

Right now - no. However, there were multiple extensions like ddterm. There are still extensions that allow you to "toggle" terminal apps.

Also, there are so many terminal apps with so similar icons, that IMO any difference, even slight, is a plus.

> Avoid metaphors that ...

I also have to remind you this is not a GNOME app, just an external project that integrates into the environment. Following the guidelines is nice, but it's not absolutely required. ddterm already does some things that go against GNOME HIG.

> and ofc we can make text even bigger but at some point it just stops looking good

![Image](https://github.com/user-attachments/assets/ad52233b-8865-4463-9350-aa4496d8003b)

IMO this still looks fine (and would look even better if `d` letters weren't looking like they're from Comic Sans - I guess the "circle" part of `d` should likely be thinner, or maybe it should be a rounded rectangle instead of circle/ellipse - for a "monospaced" font).

But this icon is already much better than a generic terminal icon.

Could you create a pull request that adds the .svg file into https://github.com/ddterm/gnome-shell-extension-ddterm/tree/master/data? You should be able to do it with "Add file" button, completely within GitHub UI (maybe you'll have to "fork" the repository first). I'd really like to record your authorship of the icon in Git history.

**Comment by thenameisluk** [comment]
> I also have to remind you this is not a GNOME app, just an external project that integrates into the environment. Following the guidelines is nice, but it's not absolutely required. ddterm already does some things that go against GNOME HIG.

if you say so

> Could you create a pull request that adds the .svg file into https://github.com/ddterm/gnome-shell-extension-ddterm/tree/master/data? You should be able to do it with "Add file" button, completely within GitHub UI (maybe you'll have to "fork" the repository first). I'd really like to record your authorship of the icon in Git history.

don't worry i know how github works
will do once we land on final logo so the pr isn't compilation of commits swapping out different parts of the logo

> And nothing is final, the icon can be adjusted/changed later too

sure

here is adwaita font

![Image](https://github.com/user-attachments/assets/edba3c56-b359-4a64-8e8e-ef609f1acce3)

and ubuntu

![Image](https://github.com/user-attachments/assets/43b8b77d-8f72-4c0f-aa4f-f4f1a19a4f1e)

are these any better?

**Comment by amezin** [comment]
Yes, I think they both do look better (and Ubuntu is likely a bit better than Adwaita) - but I'm not sure about copyright/license terms of these fonts. Drawing something, instead of using an existing font, should be safer from licensing/copyright point of view. I doubt anyone would sue me or you over using one letter in an icon of a free app - but it's still better to avoid such questions if possible :)

**Comment by amezin** [comment]
Well I'm not a lawyer too and wouldn't want to pay for a consultation. As I've said, I doubt anyone would sue us over one letter used for a free app, so I guess you can do whatever you find more convenient or interesting for you - either reuse Ubuntu font, or design your own letter `d` (if I had any ambitions as a designer, I'd definitely go for the latter though) :)

**Comment by thenameisluk** [comment]
does this work?

![Image](https://github.com/user-attachments/assets/8e66cc71-aad4-4de3-9914-1612a7244519)

**Comment by amezin** [comment]
Better than everything else except Ubuntu font. That one still looks a bit better IMO. But that's just my favorite font. This one is completely fine too.

**Comment by amezin** [comment]
No

**Comment by thenameisluk** [comment]
here you go https://github.com/ddterm/gnome-shell-extension-ddterm/pull/1285

**Comment by amezin** [comment]
Thank you for all your effort. I feel a bit guilty for being so picky - but it's the level of quality that I try to apply to my own work too.

Also thanks for completing Polish translation.

I'll open a separate issue for the symbolic icon later.

**Comment by thenameisluk** [comment]
> Thank you for all your effort. I feel a bit guilty for being so picky - but it's the level of quality that I try to apply to my own work too.

it's fine
i didn't have anything to do either way

**Comment by thenameisluk** [comment]
also if you ever want to generate nightly variant of the icon
here is the source svg that can be opened with icon preview app
[com.ddterm.it8.svg.gz](https://github.com/user-attachments/files/20093947/com.ddterm.it8.svg.gz)

**Comment by amezin** [comment]
There's currently no `.desktop` file for the development version of the app - so no way to associate an icon with it too. If you have any spare time and energy - making a symbolic icon (for the top bar/panel) is likely more important

**Comment by thenameisluk** [comment]
i came up with this

![Image](https://github.com/user-attachments/assets/d16faf0d-dc4f-4260-9273-866807e8397c)

![Image](https://github.com/user-attachments/assets/227363c2-0f3f-4392-a935-7ad7d8747d6a)

based on https://github.com/ddterm/gnome-shell-extension-ddterm/issues/724#issuecomment-2860570480
and terminal sybolic

but idk it looks good when it's small

![Image](https://github.com/user-attachments/assets/9433df76-637a-4034-9f49-ca3c3966241e)
